### PR TITLE
Show progress in machines formspec

### PR DESCRIPTION
**Pull request by asakul**
default:furnace shows progress when processing something, so I figured I'd just implement it in the same way.

### CODE REVIEW

File: technic/machines/register/machine_base.lua [hunk 1]
```diff
@@ -43,25 +43,29 @@ function technic.register_base_machine(data)
...
-		"listring[current_name;src]"..
-		"listring[current_player;main]"
-	if data.upgrade then
-		formspec = formspec..
-			"list[current_name;upgrade1;1,3;1,1;]"..
-			"list[current_name;upgrade2;2,3;1,1;]"..
-			"label[1,4;"..S("Upgrade Slots").."]"..
-			"listring[current_name;upgrade1]"..
+	local formspec = function(data, progress_percentage)
+		local formspec_string = "invsize[8,9;]"..
```
Comment (DS-Minetest) [inline technic/machines/register/machine_base.lua]: Why the indent here?

File: technic/machines/register/machine_base.lua [hunk 2]
```diff
@@ -43,25 +43,29 @@ function technic.register_base_machine(data)
...
-		"listring[current_player;main]"..
-		"listring[current_name;src]"..
-		"listring[current_player;main]"
-	if data.upgrade then
-		formspec = formspec..
-			"list[current_name;upgrade1;1,3;1,1;]"..
-			"list[current_name;upgrade2;2,3;1,1;]"..
-			"label[1,4;"..S("Upgrade Slots").."]"..
-			"listring[current_name;upgrade1]"..
+	local formspec = function(data, progress_percentage)
```
Comment (DS-Minetest) [inline technic/machines/register/machine_base.lua]: local function formspec(data, progress_percentage)

File: technic/machines/register/machine_base.lua (hunk 1 above)
Comment (DS-Minetest) [inline technic/machines/register/machine_base.lua]: `invsize` is deprecated, use `size`.

File: technic/machines/register/machine_base.lua [hunk 3]
```diff
@@ -73,6 +77,14 @@ function technic.register_base_machine(data)
 		local machine_node      = "technic:"..ltier.."_"..machine_name
 		local machine_demand    = data.demand
 
+		local update_formspec = function(result)
```
Comment (DS-Minetest) [inline technic/machines/register/machine_base.lua]: Like above. `local function update_formspec(result)`

File: technic/machines/register/machine_base.lua [hunk 4]
```diff
@@ -73,6 +77,14 @@ function technic.register_base_machine(data)
...
+			if result ~= nil then
```
Comment (DS-Minetest) [inline technic/machines/register/machine_base.lua]: Just `if result then` since it's a table.

File: technic/machines/register/machine_base.lua [hunk 5]
```diff
@@ -43,25 +43,29 @@ function technic.register_base_machine(data)
...
+			"list[current_player;main;0,5;8,4;]"..
+			"label[0,0;"..machine_desc:format(tier).."]"..
+			"listring[current_name;dst]"..
 			"listring[current_player;main]"..
-			"listring[current_name;upgrade2]"..
-			"listring[current_player;main]"
+			"listring[current_name;src]"..
+			"listring[current_player;main]"..
+			"image[3.75,1.5;1,1;gui_furnace_arrow_bg.png^[lowpart:"..
+			(progress_percentage)..":gui_furnace_arrow_fg.png^[transformR270]"
```
Comment (DS-Minetest) [inline technic/machines/register/machine_base.lua]: Don't use the parenthesis(`(`...`)`).
(http://dev.minetest.net/Lua_code_style_guidelines#Misc 4th point)

**Comment by DS-Minetest** [comment]
Actually this removes that new switch that pipeworks adds.
Also it doesn't make much sense for the furnaces since they are much faster than the arrow can be.
And could you also add this for the fuel fired alloy furnace?

**Comment by asakul** [comment]
Yeah, it updates only once per second, furnaces just work too fast. However, I didn't find any way to make smooth animation in formspec.
Also, pipeworks switch is left as it is in fuel-fired furnace, and I can't see any switch in lv-furnace, even when using latest master (both pipeworks and technic).

**Comment by DS-Minetest** [comment]
Yes, LV machines have no such switch since they are not connectable with tubes.

**Comment by asakul** [comment]
Oh, you're right, it breaks them. Have to think of something ;)

File: technic/machines/register/machine_base.lua [hunk 6]
```diff
@@ -43,28 +43,44 @@ function technic.register_base_machine(data)
...
+			formspec_string = formspec_string..
+				"list[current_name;upgrade1;1,3;1,1;]"..
+				"list[current_name;upgrade2;2,3;1,1;]"..
+				"label[1,4;"..S("Upgrade Slots").."]"..
+				"listring[current_name;upgrade1]"..
+				"listring[current_player;main]"..
+				"listring[current_name;upgrade2]"..
+				"listring[current_player;main]"
+		end
+		if tier ~= "LV" then
```
Comment (DS-Minetest) [inline technic/machines/register/machine_base.lua]:
```
if tier == "LV" then
	return formspec_string
end
.
.
.
```
Saves an indent.

**Comment by numberZero** [comment]
Wouldn’t this make large (automated) systems (like my 34-centrifuge uranium enrichment facility) more laggy?
I think, that should be done with CSM. That would lag less, and could be much smoother that way.

**Comment by asakul** [comment]
Yeah, looks like it scales pretty bad. Have to think of something again =)

**Pull request closed without merging**